Tools/Scripts/libraries/webkitcorepy/webkitcorepy/subprocess_utils.py

```diff
@@ -146,3 +147,32 @@ def decode(string):
         if check and retcode:
             raise subprocess.CalledProcessError(retcode, popenargs[0], output=decode(stdout), stderr=decode(stderr))
         return CompletedProcess(popenargs[0], retcode, decode(stdout), decode(stderr))
+
+
+class Thread(threading.Thread):
+    @classmethod
+    def terminated(cls):
+        return getattr(threading.current_thread(), '_terminated', False)
+
+    def __init__(self, *args, **kwargs):
+        super(Thread, self).__init__(*args, **kwargs)
+        self._terminated = False
+
+    def poll(self):
+        return None if self.is_alive() else {True: 1, False: 0}.get(self._terminated, -1)
+
+    def terminate(self):
+        self._terminated = True
```

Comment:
Would this be sufficient to kill the thread?

Reply:
No.

There isn't really a platform agnostic way to kill a thread. This code relies on the the thread being "killed" looking this flag (which is exactly what the unit tests do). I thought that was the best way to match the API without creating something complicated, which ultimately, we don't need (all we need is a way for the parent thread to stop an infinite loop its worker threads)